Curated Package Documentation (#1893)
aws/eks-anywhere: docs/content/en/docs/tasks/packages/harbor/_index.md

1. Generate the package configuration
   ```bash
   eksctl anywhere generate package harbor --source cluster -d .
   ```

1. Add the desired configuration to `curated-packages/my-harbor.yaml` 

   Please see [complete configuration options]({{< relref "../../../reference/packagespec/harbor" >}}) for all configuration options and their default values.

   {{% alert title="Important" color="warning" %}}
   * All configuration options are listed in dot notations (e.g., `expose.tls.enabled`) in the doc, but they have to be transformed to **hierachical structures** when specified in the `config` section in the YAML spec.
   * Harbor web portal is exposed through `NodePort` by default, and its default port number is `30003` with TLS enabled and `30002` with TLS disabled.
   * TLS is enabled by default for connections to Harbor web portal, and a secret resource named `harbor-tls-secret` is required for that purpose. It can be provisioned through cert-manager or manually with the following command using self-signed certificate:
      ```bash
      kubectl create secret tls harbor-tls-secret --cert=[path to certificate file] --key=[path to key file] -n eksa-packages
      ```

>    * `secretKey` has to be set as a string of 16 chars for encryption.

Reviewer: ```suggestion
   * `secretKey` has to be set as a string of 16 characters for encryption.
```

---
title: "Harbor package"
linkTitle: "Add Harbor"
weight: 13
date: 2022-04-12
description: >
  Install/upgrade/uninstall Harbor
---

{{% alert title="Important" color="warning" %}}

> To install package controller, please follow the [installation guide.]({{< relref ".." >}})

Reviewer 2: I don't think this needs to be an alert 
```suggestion
If your cluster was created with a release of EKS Anywhere prior to 0.9.0, you may need to [installation the package controller.]({{< relref ".." >}})
```

{{% /alert %}}

## Install

<!-- this content needs to be indented so the numbers are automatically incremented -->
1. Generate the package configuration
   ```bash

>    eksctl anywhere generate package Harbor --source cluster -d .

Reviewer 2: ```suggestion
   eksctl anywhere generate package harbor --source cluster -d .
```
This should be all case insensitive
